src/controllers/class.controller.js

```diff
@@ -0,0 +1,22 @@
+const httpStatus = require('http-status');
+const ApiError = require('../utils/ApiError');
+const catchAsync = require('../utils/catchAsync');
+const { classService } = require('../services');
+
+const createClass = catchAsync(async (req, res) => {
+  const _class = await classService.createClass({...req.body});
+  res.send({_class});
+});
+
+const getClass = catchAsync(async (req, res) => {
+  const _class = await classService.getClassById(req.class._id);
```

Comment:
once we added the `teacher` property we will need to make sure that the class belongs to the teacher here before returning it.

Reply:
i think that we don't need teacher variable in class but class variable in teacher(user). classes do not need to know who they belong to. but teachers need to know which classes they have.

Reply:
But then it becomes more difficult to validate if a class belongs to a teacher or not. If a teacher tries to access a class which is not one of his classes we need to return an error. If the teacher is stored inside the class we can just try to find it like
`Class.findOne({ _id: req.params.class_id, teacher: req.user._id })`